main.py

```diff
@@ -1,65 +1,340 @@
+                else:
+                    raise Exception('`{}` not a supported index file type'.format(_type))
+
+@click.command()
+@click.option('--plates-file')
+@click.option('--ticket-numbers-file')
+@click.option('--centriod-file')
+@click.option('--latlon-input/--no-latlon-input', is_flag=True, default=True)
+@click.option('--deduplicate/--no-deduplicate', is_flag=True, default=False, help="Add 'E' to the end of duplicate ticket numbers")
+def main(plates_file, ticket_numbers_file, centriod_file, latlon_input, deduplicate):
+    global logger, plates, plates_counter, ticket_numbers, ticket_numbers_counter, centriods, dedup
+
+    logger = get_logging()
+
+    dedup = deduplicate
+
+    plates = load_index_file(plates_file, 'plates_file')
+    plate_numbers_values = plates.values()
+    if len(plate_numbers_values) > 0:
+        plates_counter = max(map(lambda x: int(x['id']), plate_numbers_values))
     else:
-        return issuing_agency
-
-# Main
-table = etl.fromtext(strip=False)\
-            .rowmap(unpack_line, header=header, failonerror=True)\
-            .convert('fine', float)\
-            .addfield('issue_date_and_time', create_date_time)\
-            .select('{issue_date_and_time} >= "2012-01-01" and {fine} > 0')\
-            .convert('division', remove_ppa_division, pass_row=True)\
-            .convert('issuing_agency', fix_prisons_agency)
-print(etl.dicts(table))
+        plates_counter = 0
+
+    logger.info('Plates autoincrement starting at: {}'.format(plates_counter))
+
+    ticket_numbers = load_index_file(ticket_numbers_file, 'license_file')
+    ticket_numbers_values = ticket_numbers.values()
+    if len(ticket_numbers_values) > 0:
+        ticket_numbers_counter = max(map(lambda x: int(x), ticket_numbers_values))
+    else:
+        ticket_numbers_counter = 0
+
+    logger.info('Ticket number autoincrement starting at: {}'.format(ticket_numbers_counter))
+
+    centriods = load_index_file(centriod_file, 'centriod_file')
+
+    (
+        petl
+        .fromtext(strip=False)
+        .rowmap(get_transform_row(latlon_input), header=headers, failonerror=True)
+        .select('{fine} > 0.0 and {issue_datetime} >= 2012-01-01T00:00:00')
+        .rowmap(anonymize, header=headers, failonerror=True)
+        .rowmap(geocode, header=headers, failonerror=True)
+        .tocsv()
+    )
+
+    logger.info('Geocode stats - success rate: {:.2%}, successes: {}, gps: {}, zip: {}, failed_segment: {}, failed_address: {}'.format(
+        geocode_stats['success'] / geocode_stats['total'],
+        geocode_stats['success'],
+        geocode_stats['gps'],
+        geocode_stats['zip'],
+        geocode_stats['failed_segment'],
+        geocode_stats['failed_address']))
+
+    if len(geocode_stats['failed_segments']) > 0:
+        logger.info('Failed Segments - {}'.format(','.join(geocode_stats['failed_segments'])))
+
+    save_index_file(plates_file,
```

Comment:
I understand what this would do if I ran it on my local machine, but when we run it in taskflow, where does it get saved to?

Reply:
Added to docs